File: backend/app/api/schemas/usuarios.py
from pydantic import BaseModel, ConfigDict

from app.api.schemas.autenticacao import UsuarioResponse
from app.domain.enums.perfil_acesso import PerfilAcesso


class AlterarPerfilRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    perfil_acesso: PerfilAcesso


class AlterarStatusRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    ativo: bool


class ListaUsuariosResponse(BaseModel):
    usuarios: list[UsuarioResponse]
    pagina: int
    itens_por_pagina: int
    total: int
    total_paginas: int

File: frontend/src/types/usuarios.ts
import type { PerfilAcesso } from "./autenticacao";

export interface UsuarioAdministracao {
  id: string;
  nome: string;
  email: string;
  perfil_acesso: PerfilAcesso;
  equipe_id: string | null;
  ativo: boolean;
}

export interface ListaUsuariosResponse {
  usuarios: UsuarioAdministracao[];
  pagina: number;
  itens_por_pagina: number;
  total: number;
  total_paginas: number;
}

export interface AlterarPerfilUsuarioRequest {
  perfil_acesso: PerfilAcesso;
}

export interface AlterarStatusUsuarioRequest {
  ativo: boolean;
}
